### PR TITLE
Add boundary check before memset, memcpy, strncpy

### PR DESCRIPTION
**Pull request by maipbui**
#### Why I did it
Add boundary check before memset, memcpy, strncpy calls to prevent buffer overflow
##### Work item tracking
- Microsoft ADO **(number only)**: 27008041

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [x] 202106
- [x] 202111
- [x] 202205
- [x] 202211
- [x] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: src/iccpd/src/iccp_cli.c [hunk 1]
```diff
@@ -117,7 +117,17 @@ int set_peer_link(int mid, const char* ifname)
                        csm->mlag_id, ifname);
     }
 
+    if (MAX_L_PORT_NAME > strlen(csm->peer_itf_name))
```
Comment (gechiang) [inline src/iccpd/src/iccp_cli.c:120]: is this check correct?  I thought the protection is to ensure that the name is not greater than what is the allowed max value?  but the way it is checking it seems to be the other way around?  can you please check again this logic?

File: src/iccpd/src/iccp_cli.c [hunk 2]
```diff
@@ -117,7 +117,17 @@ int set_peer_link(int mid, const char* ifname)
...
+    {
+        ICCPD_LOG_ERR(__FUNCTION__, "MAX=%d is greater than peer_itf_name length=%d", MAX_L_PORT_NAME, strlen(csm->peer_itf_name));
+        return MCLAG_ERROR;
+    }
     memset(csm->peer_itf_name, 0, MAX_L_PORT_NAME);
+    if (len > strlen(csm->peer_itf_name))
```
Comment (gechiang) [inline src/iccpd/src/iccp_cli.c:126]: same concern here...

File: src/iccpd/src/iccp_cli.c [hunk 3]
```diff
@@ -208,8 +218,18 @@ int set_local_address(int mid, const char* addr)
 
     len = strlen(addr);
     memset(csm->sender_ip, 0, INET_ADDRSTRLEN);
+    if (len > strlen(csm->sender_ip))
```
Comment (gechiang) [inline src/iccpd/src/iccp_cli.c:221]: this needs to check not exceed INET_ADDRSTRLEN instead?

File: src/iccpd/src/iccp_cli.c [hunk 4]
```diff
@@ -208,8 +218,18 @@ int set_local_address(int mid, const char* addr)
...
+    {
+        ICCPD_LOG_ERR(__FUNCTION__, "len=%d is greater than sender_ip length=%d", len, strlen(csm->sender_ip));
+        return MCLAG_ERROR;
+    }
     memcpy(csm->sender_ip, addr, len);
     memset(csm->iccp_info.sender_name, 0, INET_ADDRSTRLEN);
+    if (len > strlen(csm->iccp_info.sender_name))
```
Comment (gechiang) [inline src/iccpd/src/iccp_cli.c:228]: this one needs to check against INET_ADDRSTRLEN?

File: src/iccpd/src/iccp_cli.c [hunk 5]
```diff
@@ -268,6 +288,11 @@ int set_peer_address(int mid, const char* addr)
     }
 
     memset(csm->peer_ip, 0, INET_ADDRSTRLEN);
+    if (len > strlen(csm->peer_ip))
```
Comment (gechiang) [inline src/iccpd/src/iccp_cli.c:291]: so is this check... INET_ADDRSTRLEN

File: src/iccpd/src/iccp_cli.c [hunk 6]
```diff
@@ -117,7 +117,12 @@ int set_peer_link(int mid, const char* ifname)
                        csm->mlag_id, ifname);
     }
 
-    memset(csm->peer_itf_name, 0, MAX_L_PORT_NAME);
+    memset(csm->peer_itf_name, 0, IFNAMSIZ);
+    if (len > MAX_L_PORT_NAME)
```
Comment (gechiang) [inline src/iccpd/src/iccp_cli.c:121]: this also needs to change to check for IFNAMSIZ

File: src/iccpd/src/iccp_cli.c [hunk 7]
```diff
@@ -208,8 +213,18 @@ int set_local_address(int mid, const char* addr)
 
     len = strlen(addr);
     memset(csm->sender_ip, 0, INET_ADDRSTRLEN);
+    if (len > INET_ADDRSTRLEN)
```
Comment (gechiang) [inline src/iccpd/src/iccp_cli.c:216]: for this one, did you confirm that the csm->sender_ip is created with the same "INET_ADDRSTRLEN" size, if not, you will need to change accordingly... if it is then it should be fine.

File: src/iccpd/src/iccp_cli.c (hunk 6 above)
Comment (maipbui) [reply src/iccpd/src/iccp_cli.c]: fixed

File: src/iccpd/src/iccp_cli.c (hunk 7 above)
Comment (maipbui) [reply src/iccpd/src/iccp_cli.c:216]: yes, I checked

**Review by gechiang — APPROVED** [review]

**Pull request merged**